helm/tweek/templates/infra.yaml

```diff
@@ -0,0 +1,195 @@
+  tweek_ssh_private_key: {{ .Values.secrets.tweek_ssh_private_key }}
+  minio_secret_key: {{ .Values.secrets.minio_secret_key }}
+---
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: tweek-redis-{{ .Release.Name }}
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: tweek-redis-{{ .Release.Name }}
+    spec:
+      containers:
+      - name: redis
+        image: redis:alpine
+        args: ["--appendonly", "yes"]
+        ports:
+        - containerPort: 6379
+        volumeMounts:
+        - mountPath: /data
+          name: tweek-dbvolume-{{ .Release.Name }}
+      volumes:
+      - name: tweek-dbvolume-{{ .Release.Name }}
+        persistentVolumeClaim:
+          claimName: tweek-dbvolume-{{ .Release.Name }}
+---
+kind: Service
+apiVersion: v1
+metadata:
+  name: tweek-redis-{{ .Release.Name }}
+spec:
+  selector:
+    app: tweek-redis-{{ .Release.Name }}
+  ports:
+    - port: 6379
+      targetPort: 6379
+---
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+  name: tweek-dbvolume-{{ .Release.Name }}
+spec:
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: {{ .Values.dbvolume.size }}
+  storageClassName: {{ .Values.dbvolume.storageClass }}
+---
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: tweek-minio-{{ .Release.Name }}
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: tweek-minio-{{ .Release.Name }}
+    spec:
+      containers:
+      - name: minio
+        image: minio/minio
+        args:
+        - server
+        - /data
+        env:
+        - name: MINIO_ACCESS_KEY
+          value: AKIAIOSFODNN7EXAMPLE
+        - name: MINIO_SECRET_KEY
+          valueFrom:
+            secretKeyRef:
+              name: tweek-secrets-{{ .Release.Name }}
+              key: minio_secret_key
+        ports:
+        - containerPort: 9000
+
+---
+kind: Service
+apiVersion: v1
+metadata:
+  name: tweek-minio-{{ .Release.Name }}
+spec:
+  selector:
+    app: tweek-minio-{{ .Release.Name }}
+  ports:
+    - port: 9000
+      targetPort: 9000
+---
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: tweek-nats-{{ .Release.Name }}
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: tweek-nats-{{ .Release.Name }}
+    spec:
+      containers:
+      - name: nats
+        image: nats:1.0.4-linux
+        ports:
+        - containerPort: 4222
+
+---
+kind: Service
+apiVersion: v1
+metadata:
+  name: tweek-nats-{{ .Release.Name }}
+spec:
+  selector:
+    app: tweek-nats-{{ .Release.Name }}
+  ports:
+    - port: 4222
+      targetPort: 4222
+---
+
+{{- if eq .Values.gitRepo "" -}}
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: tweek-git-{{ .Release.Name }}
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: tweek-git-{{ .Release.Name }}
+    spec:
+      containers:
+      - name: tweek-bare-repo
```

Comment:
Do we need Tweek bare repo? couldn't it be just image with git+ssh?

Reply:
Also, git might be external as well

Reply:
External git is implemented, and tweek bare repo was more straight forward to implement here that git+ssh.